services/eventservice/src/data_manager.rs

```diff
@@ -0,0 +1,301 @@
+                    })
+                    .collect()),
+                _ => Ok(match (organizer, category) {
+                    (Some(org), Some(cat)) => {
+                        let o_ids: HashSet<u64> =
+                            self._events_for_organizer(org)?.into_iter().collect();
+                        let c_ids: HashSet<u64> =
+                            self._events_for_category(cat)?.into_iter().collect();
+                        let res: Vec<u64> = (&o_ids & &c_ids).into_iter().collect();
+                        res.into_iter()
+                    }
+                    (Some(org), None) => self._events_for_organizer(org)?.into_iter(),
+                    (None, Some(cat)) => self._events_for_category(cat)?.into_iter(),
+                    (None, None) => Vec::new().into_iter(),
+                }
+                .filter_map(|v| events.get(v.to_be_bytes()).ok())
+                .filter_map(|v| v)
+                .filter_map(|v| Event::from_json(&String::from_utf8(v.to_vec()).unwrap()).ok())
+                .collect()),
+            };
+        let mut result = return_events?;
+        result.sort_by(|a, b| {
+            EventManager::_compare_by_val(sort_key, a, b).unwrap_or(std::cmp::Ordering::Equal)
+        });
+        Ok(result)
+    }
+
+    pub fn create_category(&self, cat: &Category) -> Result<(), Box<dyn std::error::Error>> {
+        let categories = self.db.open_tree(&self.categories_name)?;
+        if !categories.contains_key(&cat.as_bytes())? {
+            categories.insert(&cat.as_bytes(), b"")?;
+        }
+        Ok(())
+    }
+
+    pub fn get_categories(&self) -> Result<Vec<Category>, Box<dyn std::error::Error>> {
+        let categories = self.db.open_tree(&self.categories_name)?;
+        Ok(categories
+            .iter()
+            .filter_map(|f| f.ok())
+            .filter_map(|f| String::from_utf8(f.0.to_vec()).ok())
+            .collect())
+    }
+
+    pub fn delete_category(&self, cat: &Category) -> Result<(), Box<dyn std::error::Error>> {
+        let categories = self.db.open_tree(&self.categories_name)?;
+        if categories.contains_key(&cat.as_bytes())? {
+            categories.remove(&cat.as_bytes())?;
+        } else {
+            return Err(Box::new(std::io::Error::new(
+                std::io::ErrorKind::NotFound,
+                "Category not found",
+            )));
+        }
+        Ok(())
+    }
+
+    pub fn merge_categories(
+        &self,
+        cat1: &Category,
+        cat2: &Category,
+    ) -> Result<(), Box<dyn std::error::Error>> {
+        let categories = self.db.open_tree(&self.categories_name)?;
+        let id1 = cat1.as_bytes();
+        let id2 = cat2.as_bytes();
+        let mut to_add: Vec<u8> = Vec::new();
+        if categories.contains_key(id2)? {
+            match categories.get(id2)? {
+                Some(values) => {
+                    to_add.append(&mut values.to_vec());
+                }
+                None => {
```

Comment:
You don't need this, you already checked that the value exists. 

You can use `and_then`, `unwrap` or `map` instead of match

Reply:
Removed check above